src/silx/resources/__init__.py

```diff
@@ -52,28 +52,29 @@
              options={'py2app': {'packages': ['silx']}}
              )
 """
+from __future__ import annotations
 
 __authors__ = ["V.A. Sole", "Thomas Vincent", "J. Kieffer"]
 __license__ = "MIT"
 __date__ = "08/03/2019"
 
 
+import atexit
+import contextlib
+import importlib
+import importlib.resources
+import logging
 import os
 import sys
-import logging
-import importlib
+from typing import NamedTuple, Optional
 
 
 logger = logging.getLogger(__name__)
 
 
-# pkg_resources is useful when this package is stored in a zip
-# When pkg_resources is not available, the resources dir defaults to the
-# directory containing this module.
-try:
-    import pkg_resources
-except ImportError:
-    pkg_resources = None
+# Manage resource files life-cycle
+_FILE_MANAGER = contextlib.ExitStack()
+atexit.register(_FILE_MANAGER.close)
```

Comment:
See https://importlib-resources.readthedocs.io/en/latest/migration.html#pkg-resources-resource-filename for explanation